src/scanner/sky_test.go

```diff
@@ -0,0 +1,502 @@
+package scanner
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/skycoin/teller/src/util/dbutil"
+	"github.com/skycoin/teller/src/util/testutil"
+	"github.com/stretchr/testify/require"
+
+	"github.com/skycoin/skycoin/src/visor"
+	"github.com/skycoin/skycoin/src/visor/blockdb"
+)
+
+var (
+	errNoSkyBlockHash = errors.New("no block found for height")
+)
+
+type dummySkyrpcclient struct {
+	db                           *bolt.DB
+	blockHashes                  map[int64]string
+	blockCount                   int64
+	blockCountError              error
+	blockVerboseTxError          error
+	blockVerboseTxErrorCallCount int
+	blockVerboseTxCallCount      int
+
+	// used for testBtcScannerBlockNextHashAppears
+	blockNextHeightMissingOnceAt uint64
+	hasSetMissingHeight          bool
+}
+
+func openDummySkyDB(t *testing.T) *bolt.DB {
+	// Blocks 0 through 180 are stored in this DB
+	db, err := bolt.Open("./sky.db", 0600, nil)
+	require.NoError(t, err)
+	return db
+}
+
+func newDummySkyrpcclient(db *bolt.DB) *dummySkyrpcclient {
+	return &dummySkyrpcclient{
+		db:          db,
+		blockHashes: make(map[int64]string),
+	}
+}
+
+func (dsc *dummySkyrpcclient) GetBlockCount() (int64, error) {
+	if dsc.blockCountError != nil {
+		// blockCountError is only returned once
+		err := dsc.blockCountError
+		dsc.blockCountError = nil
+		return 0, err
+	}
+
+	return dsc.blockCount, nil
+}
+
+func (dsc *dummySkyrpcclient) GetBlockVerboseTx(seq uint64) (*visor.ReadableBlock, error) {
+	//TODO (therealssj): refactor this to directly read from the database
```

Comment:
I dont have a good understanding of the skycoin blockchain structure so I just used the existing functions instead of reading the database values directly